src/common/trapezoidal.jl

```diff
@@ -0,0 +1,16 @@
+# NumericalIntegration.jl also implements this (in a slightly different way)
+# however, it is unmaintained and has stale, conflicting version requirements
+# for Interpolations.jl
+@inbounds function trapezoidal(x::AbstractVector, y::AbstractVector)
+    n = length(x)
+    n == length(y) || error("vectors `x` and `y` must have the same number of elements")
```

Comment:
I think I would even make this an assert.

Reply:
I may be misremembering this, but I think `@assert` can be dropped by the compiler.

Reply:
You can actually mark such checks as "the bounds checks" using a macro. I forgot how it works, but I suppose that's the way to go here.